src/transformState.js

```diff
@@ -5,7 +5,44 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(state, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clear(state);
+        break;
+
+      default:
+        throw new Error("Something went wrond! Please check your data!");
+    }
+  }
+}
+
+function addProperties(state, data) {
+  for (const key in data) {
+    state[key] = data[key];
+  }
+}
+
+function removeProperties(state, data) {
+  for (const key of data) {
+    if (state[key] !== undefined) {
+      delete state[key];
+    }
+  }
+}
+
+function clear(state) {
+  for (const key in state) {
+    delete state[key];
+  }
```

Comment:
Well done, approving! Very clean code! You extracted actions logic into separated functions - my respect) For me it's better if this functions were imported from separate file, but i'm not sure if it not to much for this little task)